Mathlib/LinearAlgebra/Matrix/LinearProgramming.lean

```diff
@@ -0,0 +1,63 @@
+/-
+Copyright (c) 2024 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak
+-/
+import Mathlib.LinearAlgebra.Matrix.DotProduct
+
+/-!
+
+# Linear programming
+
+TODO
+
+## Main definitions
+
+ * `StandardLP` defines a linear program in the standard form.
+ * `StandardLP.Admits` tells if given vector is an admissible solution to given standard LP.
+
+-/
+
+open Matrix
+
+/-- Linear program in the standard form. -/
+structure StandardLP (m n K : Type) [Fintype m] [Fintype n] [LinearOrderedField K] where
+  /-- Matrix of coefficients -/
+  A : Matrix m n K
+  /-- Right-hand side -/
+  b : m → K
+  /-- Objective function coefficients -/
+  c : n → K
+
+variable {m n K : Type} [Fintype m] [Fintype n] [LinearOrderedField K]
+
+def StandardLP.Admits (P : StandardLP m n K) (x : n → K) : Prop :=
+  P.A.mulVec x ≤ P.b ∧ 0 ≤ x
+
+def StandardLP.Admissible (P : StandardLP m n K) : Prop :=
+  ∃ x : n → K, P.Admits x
+
+def StandardLP.Reaches (P : StandardLP m n K) (v : K) : Prop :=
```

Comment:
Again, I haven't seen this usage of `Reaches` but I don't have an alternate suggestions.